dali/base/dasds.cpp

```diff
@@ -5760,45 +5761,81 @@ class CStoreHelper : implements IStoreHelper, public CInterface
             constexpr const char *temporaryXmlFileSaveName = "dali_store_tmp.xml";
             OwnedIFileIO iXmlFileIOTmpStore = createUniqueFile(location, temporaryXmlFileSaveName, NULL, tmpStoreName);
             OwnedIFile iXmlFileTmpStore = createIFile(tmpStoreName);
-            unsigned xmlCrc{0};
-            try
-            {
-                xmlCrc = saveStoreToFile(root, iXmlFileIOTmpStore, StoreFormat::XML);
-            }
-            catch (IException *e)
-            {
-                StringBuffer errMsg;
-                EXCLOG(e, errMsg.append("Failed to save XML temporary store to : ").append(temporaryXmlFileSaveName).str());
-                e->Release();
-                iXmlFileTmpStore->remove();
-                return;
-            }
 
             // Save store as binary
             OwnedIFile iBinaryFileTmpStore;
+            constexpr const char *temporaryBinaryFileSaveName = "dali_store_tmp.bin";
+            OwnedIFileIO iBinaryFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
+            iBinaryFileTmpStore.setown(createIFile(tmpStoreName));
+
+            // Execute save operations
+            unsigned xmlCrc{0};
             unsigned binaryCrc{0};
-            unsigned *binaryCrcPtr{nullptr};
-            if (saveBinary)
+            bool binaryStoreSaved = false;
+
+            if (saveAsync && saveBinary)
             {
-                StringBuffer tmpStoreName;
-                constexpr const char *temporaryBinaryFileSaveName = "dali_store_tmp.bin";
-                OwnedIFileIO iBinaryFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
-                iBinaryFileTmpStore.setown(createIFile(tmpStoreName));
-                try
+                // Execute saves in parallel only if both saves are enabled
+                Owned<CCompletionTask> completed = new CCompletionTask();
```

Comment:
tasks are not really suited for this purpose, and overly complicated.
Should use jthread objects, or std::threads.